v2/pubsub-to-redis/src/main/java/com/google/cloud/teleport/v2/templates/transforms/MessageTransformation.java

```diff
@@ -0,0 +1,174 @@
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Map;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubIO;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubMessage;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.values.KV;
+import org.checkerframework.checker.initialization.qual.Initialized;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.UnknownKeyFor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Different transformations over the processed data in the pipeline. */
+public class MessageTransformation {
+  private static final Logger LOG = LoggerFactory.getLogger(MessageTransformation.class);
+
+  private static final ObjectMapper objectMapper = new ObjectMapper();
+
+  static String attributeKey;
+  static String messageId;
+  public static String key;
+  public static String hashKey;
+  public static String hashKeyPrefix = "hash:";
+  public static String logKeyPrefix = "log:";
+  public static String namespaceDelimiter = ":";
+  static String fieldKey = "data";
+
+  /** Configures Pubsub consumer. */
+  public static PubsubIO.@UnknownKeyFor @NonNull @Initialized Read<
+          @UnknownKeyFor @NonNull @Initialized PubsubMessage>
+      readFromPubSub(String inputSubscription) {
+    PubsubIO.@UnknownKeyFor @NonNull @Initialized Read<
+            @UnknownKeyFor @NonNull @Initialized PubsubMessage>
+        pubSubRecords = PubsubIO.readMessagesWithAttributesAndMessageId();
+    return pubSubRecords.fromSubscription(inputSubscription);
+  }
+
+  public static class MessageToRedisString extends DoFn<PubsubMessage, String> {
+    @ProcessElement
+    public void processElement(
+        @Element PubsubMessage pubsubMessage, OutputReceiver<String> receiver) {
+      String element = new String(pubsubMessage.getPayload());
+      messageId = pubsubMessage.getMessageId();
+      LOG.debug("PubSubMessage messageId: " + messageId);
+      LOG.debug("PubSubMessage payload: " + element);
+
+      if (pubsubMessage.getAttribute("key") != null) {
+        attributeKey = pubsubMessage.getAttribute("key");
+        LOG.debug("PubSubMessage attributeKey: " + attributeKey);
+        key = attributeKey + namespaceDelimiter + messageId;
+      } else {
+        attributeKey = "";
+        key = messageId;
+      }
+      receiver.output(element);
+    }
+  }
+
+  public static class MessageToRedisHash
+      extends DoFn<PubsubMessage, KV<String, KV<String, String>>> {
+    @ProcessElement
+    public void processElement(
+        @Element PubsubMessage pubsubMessage,
+        OutputReceiver<KV<String, KV<String, String>>> receiver) {
+      String element = new String(pubsubMessage.getPayload());
+      messageId = pubsubMessage.getMessageId();
+      LOG.debug("PubSubMessage messageId: " + messageId);
+      LOG.debug("PubSubMessage payload: " + element);
```

Comment:
It is recommended to not log any user (potentially confidential) information to logs -- can we remove those debugs?

Reply:
Removed payload debug and left the messageId.